parquet-hadoop/src/main/java/org/apache/parquet/hadoop/CodecFactory.java

```diff
@@ -234,11 +234,11 @@ protected CompressionCodec getCodec(CompressionCodecName codecName) {
     if (codecClassName == null) {
       return null;
     }
-    CompressionCodec codec = CODEC_BY_NAME.get(codecClassName);
+    String codecCacheKey = this.cacheKey(codecName);
+    CompressionCodec codec = CODEC_BY_NAME.get(codecCacheKey);
```

Comment:
Since `CODEC_BY_NAME` is `protected`, I think this _could_ break something that is relying on the cache, although I'm not sure why someone would access it directly. Maybe that visibility is an accident?

Reply:
If that is a concern, we can cache the old key (w/o level) as well.

Reply:
Personally, I'm not too worried about this, I don't see anyone doing this. At least nobody in the Apache org: https://github.com/search?q=org%3Aapache%20CODEC_BY_NAME&type=code :)